File: rhinocommon/snippets/cs/add-linear-dimension.cs
partial class Examples
{
  public static Rhino.Commands.Result AddLinearDimension(Rhino.RhinoDoc doc)
  {
    Rhino.Geometry.LinearDimension dimension;
    Rhino.Commands.Result rc = Rhino.Input.RhinoGet.GetLinearDimension(out dimension);
    if (rc == Rhino.Commands.Result.Success && dimension != null)
    {
      if (doc.Objects.AddLinearDimension(dimension) == Guid.Empty)
        rc = Rhino.Commands.Result.Failure;
      else
        doc.Views.Redraw();
    }
    return rc;
  }
}

File: rhinocommon/snippets/vbnet/add-linear-dimension.vb
Partial Friend Class Examples
  Public Shared Function AddLinearDimension(ByVal doc As Rhino.RhinoDoc) As Rhino.Commands.Result
	Dim dimension As Rhino.Geometry.LinearDimension = Nothing
	Dim rc As Rhino.Commands.Result = Rhino.Input.RhinoGet.GetLinearDimension(dimension)
	If rc Is Rhino.Commands.Result.Success AndAlso dimension IsNot Nothing Then
	  If doc.Objects.AddLinearDimension(dimension) = Guid.Empty Then
		rc = Rhino.Commands.Result.Failure
	  Else
		doc.Views.Redraw()
	  End If
	End If
	Return rc
  End Function
End Class
